src/basic/rm-rf.c

```diff
@@ -182,18 +182,11 @@ int rm_rf(const char *path, RemoveFlags flags) {
         /* We refuse to clean the root file system with this
          * call. This is extra paranoia to never cause a really
          * seriously broken system. */
-        if (path_equal(path, "/")) {
+        if (path_equal_or_files_same(path, "/")) {
```

Comment:
I figure we should a flavour of files_same() that doesn't follow symlinks, and similar a flavour of path_equal_or_files_same(), i.e. honouring a flag AT_SYMLINK_NOFOLLOW or so